### PR TITLE
Jbeck/ag 1144/transform gene info testing

### PR DESCRIPTION
**Pull request by jaclynbeck-sage**
This PR adds integration testing for the `gene_info` transform. I also fixed several bugs/missing handling in the transform that I discovered while writing the tests. 
There are 2 'passing' cases with good data, using two sets of values for `adjusted_p_value_threshold` and `protein_level_threshold`, which are specified in the config file. 
There are 10 'failure' cases where we expect the code to throw errors:
- (3) tests for duplicate Ensembl IDs in 3 of the input datasets
- (4) tests for non-numeric values in key columns from 4 input data sets
- (1) test for missing HGNC symbol in `tep_adi_info`
- (2) tests for non-boolean values in 2 fields of `tep_adi_info`

I tried to document a description of each of the data sets and what conditions would pass/fail for each one, at the top of the test_gene_info.py file. 

Unfortunately, the `gene_metadata` test files have to be in feather format, not csv, due to having lists of items inside the `alias` and `ensembl_possible_replacements` fields. So those files are not human-readable. 

Changes made to the transform itself:
- biodomains data set drops rows with NA "biodomain" values to avoid "nan" ending up in the list of biodomains for that Ensembl ID
- tep_adi_info data set now has a check to make sure all the values in the `is_adi` and `is_tep` fields are boolean (or missing, which is also ok)
- added handling for edge case where `ensembl_info` ends up as `null` instead of a dictionary of null values. Moving the `nest_fields` call for this field from **before** the merge with `gene_info` to **after all other data sets were merged**, and ensuring that all `ensembl_possible_replacements` values were a list or empty list and not `null`, solves the issue. 
- some reformatting by Black

### CODE REVIEW

File: src/agoradatatools/etl/transform/gene_info.py [hunk 1]
```diff
@@ -92,36 +93,32 @@ def transform_gene_info(
...
-        ]
-    ]
-    ensembl_info = nest_fields(
-        df=ensembl_info,
-        grouping="ensembl_gene_id",
-        new_column="ensembl_info",
-        drop_columns=["ensembl_gene_id"],
-        nested_field_is_list=False,
-    )
-
```
Comment (jaclynbeck-sage) [inline src/agoradatatools/etl/transform/gene_info.py:124]: FYI The above chunk of code has been moved further down the file.

File: src/agoradatatools/etl/transform/gene_info.py [hunk 2]
```diff
@@ -92,36 +93,32 @@ def transform_gene_info(
     # sort biodomains list alphabetically
     biodomains["biodomains"] = biodomains["biodomains"].apply(sorted)
 
+    # Type-check the 'is_adi' and 'is_tep' columns of tep_info to make sure they are booleans and not strings.
+    # Explicitly make NaN is_adi and is_tep values "False" to avoid having to check for boolean and NaN in the
+    # check below.
+    tep_info = tep_info.fillna({"is_adi": False, "is_tep": False})
+    if tep_info["is_adi"].dtype != bool:
+        raise TypeError("wrong data type in 'is_adi' column")
```
Comment (BWMac) [inline src/agoradatatools/etl/transform/gene_info.py]: NIT: You might want to make these error messages more specific

```suggestion
        raise TypeError(f"'is_adi' column must be `bool`, currently is {tep_info["is_adi"].dtype}")
```

File: tests/transform/test_gene_info.py [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+            param_set_1,
+            TypeError,
+            "can only concatenate str",
+        ),
+        (  # is_adi is a string
+            core_files,
+            {"tep_adi_info": "tep_adi_info_type_error_2.csv"},
+            param_set_1,
+            TypeError,
+            "wrong data type in 'is_adi' column",
```
Comment (BWMac) [inline tests/transform/test_gene_info.py]: Just a note that if you update the error messages from my other comment you will need to update these too

File: tests/transform/test_gene_info.py [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+        """Utility function to read a dictionary of filenames into a dictionary of data frames. Most files for
+        gene_info are in csv format, but the 'gene_metadata' file is in feather format and needs special casing.
+
+        Args:
+            input_files_dict - a dictionary where keys are the names of the datasets, as expected by
+                               transform_gene_info, and values are the filenames to load
+
+        Returns:
+            datasets - a dictionary where the keys are the names of the datasets, as expected by
+                       transform_gene_info, and the values are data frames
```
Comment (BWMac) [inline tests/transform/test_gene_info.py]:
```suggestion
        Args:
            input_files_dict: a dictionary where keys are the names of the datasets, as expected by
                               transform_gene_info, and values are the filenames to load

        Returns:
            datasets: a dictionary where the keys are the names of the datasets, as expected by
                       transform_gene_info, and the values are data frames
```

**Review by BWMac — APPROVED** [review]
LGTM! Just a couple NITs

**Review by BWMac — APPROVED** [review]

**Review by JessterB — APPROVED** [review]
LGTM!

**Pull request merged**